### PR TITLE
Added Crowdin Translation Provider

### PR DESCRIPTION
**Pull request by andrii-bodnar**
Hi!

It's a cool idea to make Translation Providers for Symfony. Great job, guys! :+1: 

Previously I helped @welcoMattic with some Crowdin API methods in the [first PR](https://github.com/symfony/symfony/pull/37462) and in chat in the Symfony Devs Slack workspace.

A few weeks ago a checked the [new PR](https://github.com/symfony/symfony/pull/38475) and realized that Crowdin Provider is missing.

So, I decided to do it by myself (of course, after @welcoMattic's approval) :slightly_smiling_face:

Therefore, this PR adds Crowdin Provider to the Translation Providers feature. It already supports both [crowdin.com](https://crowdin.com/) and [Crowdin Enterprise](https://crowdin.com/enterprise).

### CODE REVIEW

File: src/Symfony/Component/Translation/Bridge/Crowdin/Provider/CrowdinProvider.php [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+     */
+    public function delete(TranslatorBag $translatorBag): void
+    {
+        foreach($translatorBag->getDomains() as $domain) {
+            $fileId = $this->getFileId($domain);
+
+            if ($fileId) {
+                $this->deleteFile($fileId);
+            }
+        }
```
Comment (welcoMattic) [inline src/Symfony/Component/Translation/Bridge/Crowdin/Provider/CrowdinProvider.php]: Oh! This workaround will remove all translations (key + messages) of a domain. This it not that we want. Here, we want to delete only the translations contained in the TranslatorBag passed as argument. Which contains the diff, computed in the Command file.

We have to rework this method

File: src/Symfony/Component/Translation/Bridge/Crowdin/Provider/CrowdinProvider.php [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+            ]),
+        ]);
+
+        $responseContent = $response->getContent(false);
+
+        if (Response::HTTP_CREATED !== $response->getStatusCode()) {
+            throw new ProviderException(sprintf('Unable to add a File in Crowdin for domain "%s": "%s".', $domain, $responseContent), $response);
+        }
+
+        dump(sprintf('[Crowdin info] File "%s.%s" added successfully.', $domain, 'xlf'));
```
Comment (welcoMattic) [inline src/Symfony/Component/Translation/Bridge/Crowdin/Provider/CrowdinProvider.php]: `dump` is a dev only function that should be used for debug purpose only. We don't need to "display" anything here. That's the role of the Commands. Please remove all instances of `dump` call

File: src/Symfony/Component/Translation/Bridge/Crowdin/Provider/CrowdinProvider.php [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+        if (Response::HTTP_OK !== $response->getStatusCode()) {
+            throw new ProviderException(
+                sprintf('Unable to export file %d translations for language %s.', $fileId, $languageId),
+                $response
+            );
+        }
+
+        $export = json_decode($response->getContent(), true);
+
+        return file_get_contents($export['data']['url']);
```
Comment (welcoMattic) [inline src/Symfony/Component/Translation/Bridge/Crowdin/Provider/CrowdinProvider.php]: You must use the HttpClient to fetch the content of the export file here, to handle potential failure properly.

File: src/Symfony/Component/Translation/Bridge/Crowdin/Provider/CrowdinProvider.php [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+        if (Response::HTTP_OK !== $response->getStatusCode()) {
+            throw new ProviderException(
+                sprintf('Unable to download source file %d.', $fileId),
+                $response
+            );
+        }
+
+        $export = json_decode($response->getContent(), true);
+
+        return file_get_contents($export['data']['url']);
```
Comment (welcoMattic) [inline src/Symfony/Component/Translation/Bridge/Crowdin/Provider/CrowdinProvider.php]: Same here.

File: src/Symfony/Component/Translation/Bridge/Crowdin/Provider/CrowdinProvider.php [hunk 5]
```diff
@@ -0,0 +1,395 @@
...
+        /**
+         * @see https://support.crowdin.com/api/v2/#operation/api.projects.files.delete (Crowdin API)
+         * @see https://support.crowdin.com/enterprise/api/#operation/api.projects.files.delete (Crowdin Enterprise API)
+         */
+        $response = $this->client->request('DELETE', sprintf('https://%s/projects/%d/files/%d', $this->getEndpoint(), $this->projectId, $fileId), [
+            'headers' => $this->getDefaultHeaders()
+        ]);
+
+        if (Response::HTTP_NOT_FOUND === $response->getStatusCode()) {
+            dump(sprintf('[Crowdin info] File ID "%d" Not Found for the project %d.', $fileId, $this->projectId));
```
Comment (welcoMattic) [inline src/Symfony/Component/Translation/Bridge/Crowdin/Provider/CrowdinProvider.php]: Please use an Exception here, and format the error text properly (no need of `[Crowdin info]`)

**Review by welcoMattic — COMMENTED** [review]
Globally, it looks great! You have implemented all parts where I encountered difficulties :pray: Thank you!

I left some comments, could you please fix them? After what, I think I could merge your PR to move forward your Bridge into the main PR on symfony's repo!

Thanks again!

**Comment by andrii-bodnar** [comment]
@welcoMattic thanks a lot for your feedback and review! :+1: 

Sure, I will try to resolve all the discussions this weekend.

**Review by welcoMattic — APPROVED** [review]
It looks much better! Thanks a lot! (Red CI is not on your fault)

**Pull request merged**